src/support.cpp

```diff
@@ -48,6 +48,255 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    unsigned int gradual_support_step_layer_count = round_divide(gradual_support_step_height, storage.getSettingInMicrons("layer_height")); // The difference in layer count between consecutive density infill areas
+
+    // make gradual_support_step_height divisable by layer_skip_count
+    float n_skip_steps_per_gradual_step = std::max(1.0f, std::ceil(gradual_support_step_layer_count / layer_skip_count)); // only decrease layer_skip_count to make it a divisor of gradual_support_step_layer_count
+    layer_skip_count = gradual_support_step_layer_count / n_skip_steps_per_gradual_step;
+
+    size_t min_layer = 0;
+    size_t max_layer = total_layer_count - 1;
+
+    const coord_t wall_line_width_x = storage.getSettingInMicrons("wall_line_width_x");
+    const int skin_outline_count = 1;  // we only have 1 skin for support infill
+
+    for (uint32_t layer_nr = 0; layer_nr < total_layer_count - 1; ++layer_nr)
+    {
+        assert(storage.support.supportLayers[layer_nr].support_infill_part_list.empty() && "support infill part list is supposed to be uninitialized");
+
+        // this is the complete support areas on this layer
+        const Polygons& whole_support_areas = storage.support.supportLayers[layer_nr].supportAreas;
+
+        if (whole_support_areas.size() == 0 or layer_nr < min_layer or layer_nr > max_layer)
+        {
+            // initialize support_infill_part_list empty
+            storage.support.supportLayers[layer_nr].support_infill_part_list.clear();
+            continue;
+        }
+
+        // generate separate support islands and calculate density areas for each island
+        std::vector<PolygonsPart> support_islands = whole_support_areas.splitIntoParts();
+        for (uint32_t i = 0; i < support_islands.size(); ++i)
+        {
+            storage.support.supportLayers[layer_nr].support_infill_part_list.emplace_back();
+            SupportInfillPart& support_infill_part = storage.support.supportLayers[layer_nr].support_infill_part_list.back();
+            support_infill_part.gradual_infill_areas.clear();
+            support_infill_part.outline = support_islands[i];
+            support_infill_part.insets.clear();
+            // generate insets and use the first inset as the infill area
+            AreaSupport::generateOutlineInsets(
+                support_infill_part.insets,
+                support_infill_part.outline,
+                skin_outline_count,
+                wall_line_width_x);
+            assert(support_infill_part.gradual_infill_areas.size() == 0);
+            if (support_infill_part.insets.empty())
+            {
+                continue;
+            }
+
+            // calculate density areas for this island
+            Polygons less_dense_support = support_infill_part.insets[0]; // one step less dense with each support_step
+            for (unsigned int support_step = 0; support_step < max_support_steps; ++support_step)
```

Comment:
The term 'support_step' doesn't really make sense.
Suggested: `density_step` and `max_density_steps`